packages/lesswrong/server/analyticsWriter.ts

```diff
@@ -35,65 +35,39 @@ addGraphQLResolvers({
         const adjustedTimestamp = isValidEventAge(age) ? new Date(serverTime.valueOf()-age.valueOf()) : serverTime;
         
         let eventCopy = {...event, timestamp: adjustedTimestamp};
-        writeEventToAnalyticsDB(eventCopy);
+        void writeEventToAnalyticsDB(eventCopy);
```

Comment:
Can you explain to me the changes happening in this file? Up to and including what the analyticsWriter file does. I assumed data goes into the database via captureEvent, so I'm a bit confused what `analyticsWriters` is about.

Thanks.

Reply:
One note: We're not actually converting the function to be async, in the broad sense. The function was using the old-style callback method of async programming, we just switched it to the new style.

As for what `writeEventToAnalyticsDB` does, it is the basement level function underneath `captureEvent`. `captureEvent` can run on the client or server, and does nice things like record the timestamp for you. Ultimately someone needs to send the data to the db though, and that's what this function does.

Ultimately the cause of these changes are that we moved towards a library `'pg-promise'` that allows talking to the postgres database with 'modern'* promises, as opposed to old-style callback methods. To avoid having multiple `getAnalyticsConnection` functions, we needed to refactor this file to use the new library. There should be no behavioral differences to this file.

*Promises first shipped to chrome in 2014, so... '''modern'''